force-app/main/userCommunity/classes/HOT_RequestListContoller.cls

```diff
@@ -1,4 +1,48 @@
 public without sharing class HOT_RequestListContoller {
+    @AuraEnabled
+    public static void uploadFile(String base64, String filename, String recordId) {
+        ContentVersion cv = createContentVersion(base64, filename);
+        createContentLink(cv.Id, recordId);
+        setFileConsentToTrue(recordId);
+    }
+
+    private static ContentVersion createContentVersion(String base64, String filename) {
+        ContentVersion cv = new ContentVersion();
+        cv.VersionData = EncodingUtil.base64Decode(base64);
+        cv.Title = filename;
+        cv.PathOnClient = filename;
+        try {
+            insert cv;
+            return cv;
+        } catch (DMLException e) {
+            System.debug(e);
+            return null;
+        }
+    }
+
+    private static void createContentLink(String contentVersionId, String recordId) {
+        if (contentVersionId == null || recordId == null) {
+            return;
+        }
+        ContentDocumentLink cdl = new ContentDocumentLink();
+        cdl.ContentDocumentId = [SELECT ContentDocumentId FROM ContentVersion WHERE Id = :contentVersionId]
+        .ContentDocumentId;
+        cdl.LinkedEntityId = recordId;
+        cdl.ShareType = 'V'; // V = Viewer, C = Collaborator, I = Inferred
+        cdl.Visibility = 'AllUsers';
+        try {
+            insert cdl;
+        } catch (DMLException e) {
+            System.debug(e);
+        }
+    }
+
+    private static void setFileConsentToTrue(String recordId) {
```

Comment:
Kan ikke dette gjøres som en del av skjemainnsending? I javascriptet? this.fieldValues.IsFileConsent__c = true; om de godtar.